spec/classes/application_spec.rb

```diff
@@ -280,6 +286,96 @@ class {'katello::globals':
+          <<-EOS
+          class {'katello::params':
+            enable_katello_agent => true,
+            candlepin_oauth_secret => 'candlepin-secret',
+          }
+          EOS
+        end
+
+        it { is_expected.to compile.with_all_deps }
+
+        let(:katello_yaml_content) do
+          [
+            ':katello:',
+            '  :rest_client_timeout: 3600',
+            '  :content_types:',
+            '    :yum: true',
+            '    :file: true',
+            '    :deb: true',
+            '    :puppet: false',
+            '    :docker: true',
+            '    :ostree: false',
+            '  :candlepin:',
+            '    :url: https://localhost:23443/candlepin',
+            '    :oauth_key: "katello"',
+            '    :oauth_secret: "candlepin-secret"',
+            '    :ca_cert_file: /etc/pki/katello/certs/katello-default-ca.crt',
+            '  :candlepin_events:',
+            '    :ssl_cert_file: /etc/pki/katello/certs/java-client.crt',
+            '    :ssl_key_file: /etc/pki/katello/private/java-client.key',
+            '    :ssl_ca_file: /etc/pki/katello/certs/katello-default-ca.crt',
+            '  :agent:',
+            '    :enabled: true',
+            '    :broker_url: amqps://localhost:5671',
+            '    :event_queue_name: katello.agent',
+            '  :katello_applicability: true',
+          ]
+        end
+
+        it 'should generate correct katello.yaml' do
+          verify_exact_contents(catalogue, '/etc/foreman/plugins/katello.yaml', katello_yaml_content)
+        end
+      end
+
+      context 'with katello::params::enable_katello_agent empty string' do
+        let :pre_condition do
+          <<-EOS
+          class {'katello::params':
+            enable_katello_agent => '',
+            candlepin_oauth_secret => 'candlepin-secret',
+          }
+          EOS
+        end
+
+        it { is_expected.to compile.with_all_deps }
+
+        let(:katello_yaml_content) do
+          [
+            ':katello:',
+            '  :rest_client_timeout: 3600',
+            '  :content_types:',
+            '    :yum: true',
+            '    :file: true',
+            '    :deb: true',
+            '    :puppet: false',
+            '    :docker: true',
+            '    :ostree: false',
+            '  :candlepin:',
+            '    :url: https://localhost:23443/candlepin',
+            '    :oauth_key: "katello"',
+            '    :oauth_secret: "candlepin-secret"',
+            '    :ca_cert_file: /etc/pki/katello/certs/katello-default-ca.crt',
+            '  :candlepin_events:',
+            '    :ssl_cert_file: /etc/pki/katello/certs/java-client.crt',
+            '    :ssl_key_file: /etc/pki/katello/private/java-client.key',
+            '    :ssl_ca_file: /etc/pki/katello/certs/katello-default-ca.crt',
+            '  :agent:',
+            '    :enabled: false',
+            '    :broker_url: amqps://localhost:5671',
+            '    :event_queue_name: katello.agent',
```

Comment:
```suggestion
            '  :agent:',
            '    :enabled: false',
```

Should we simplify the config and omit these when the katello agent is disabled?

Reply:
I think it would be straightforward to wrap those lines with a conditional in templates/katello.yaml.erb

Reply:
Good idea, done.